### PR TITLE
Talent and Equipment Log create

### PR DESCRIPTION
**Pull request by Caleb1991**
Talent and Equipment Log create functionality, waiting to see if we want to reference the `weapon` object under `weapons` in the `equipment_log`. In addition, waiting to see if we want to update character-equipment_log relationship to be a `has_one`.

### CODE REVIEW

**Comment by Caleb1991** [comment]
Sooooooo, here's what ended up happening:

I went through and changed `weapons` on `equipment_logs` from a `string` to an `array`

I then changed `weapons` on `equipment_logs` to be `weapon_ids`

From there, I added the `validate_uniqueness_of :character_id` on `equipment_logs` so that a character may only have one `equipment_log1`

Changed the `has_many` to a `has_one` relationship between a `character` and their `equipment_log`

Figured a joins table would be much more useful than the `weapon_ids` so created `weapons_in_equipment_logs` and did a `has_many :weapons, through: :weapons_in_equipment_logs` on the `equipment_log` model

Hopefully this explains the like five migrations I made, if you'd like to do this a different way, we can totally do that!

File: app/models/character.rb [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-  validates_presence_of :name
-  validates_presence_of :species
-  validates_presence_of :specialization
-  validates_presence_of :career
-  validates_presence_of :age
-  validates_presence_of :height
-  validates_presence_of :build
-  validates_presence_of :hair
-  validates_presence_of :eyes
+  validates_presence_of  :user_id,
```
Comment (suzkiee) [inline app/models/character.rb:14]: Thanks for adding the validation!

**Review by suzkiee — COMMENTED** [review]
Sorry to be "that person" but could we get simple unit tests for the new mutations?

**Comment by Caleb1991** [comment]
> 

Thank you for calling this out! I should have already been doing tests but this will make me actually sit down and do them so thank you!

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ def user(id:)
     end
 
     # Find a list of characters for a single user
+    # Do we need both of these `:user` field declerations or can we consolidate them to one and just use def on both of them?
     field :user, Types::UserType, null: false do
       argument :id, ID, required: true
     end
```
Comment (Caleb1991) [inline app/graphql/types/query_type.rb:20]: Suzie, your expertise ples!

Comment (suzkiee) [reply app/graphql/types/query_type.rb:20]: oh yeah I think this is just duplicate code. Great catch!

**Review by suzkiee — APPROVED** [review]
Thank you for the thorough testing and your patience through the process!

**Pull request merged**